File: src/test/java/pages/DashboardPage.java
package pages;

import com.microsoft.playwright.Page;

public class DashboardPage extends BaseTest {
    private Page page;
    private String balanceLocator = "//table[@id='accountTable']/tbody/tr[2]/td[2]/b";

    public DashboardPage(Page page) {
        this.page = page;
    }

    public String getBalance() {
        page.waitForTimeout(4000);
        return page.textContent(balanceLocator);
    }
}

File: src/pages/dashboard.page.ts
import { Page } from 'playwright';

export class DashboardPage {
  private readonly balanceLocator = "//table[@id='accountTable']/tbody/tr[2]/td[2]/b";

  constructor(private readonly page: Page) {}

  async getBalance(): Promise<string> {
    await this.page.waitForTimeout(4000);
    const balance = await this.page.textContent(this.balanceLocator);
    return balance ?? '';
  }
}
